### DISCUSSION TITLE
How to call validation manually for a field or property.

### DISCUSSION THREAD

**Discussion by mishael-o** [Blazored/FluentValidation#149] [Ideas]
Hi, I have a scenario where I would want to revalidate a property when another property has changed.

Example
`DateFrom` and `DateTo` properties. When `DateFrom` changes I want to call validations for `DateTo` and vice versa to ensure their states is in sync.

I see the `FluentValidationValidator` exposes `Validate` and `ValidateAsync` however there is no API validating a property.

**Comment by Sharaf-Mansour** [comment]
@mishael-o 
Maybe you are looking for something like this
```csharp
    public  bool IsDateFromValid() => new DateValidator().Validate(DateFrom).IsValid;
```
Then call the function in the other abstract validator using Must or MustAsync 
Or you can even do something like
```Csharp
   public class FooValidator : AbstractValidator<Foo>
    {
         DateTime dateFrom , dateTo;
         publicFooValidator()
         {
            RuleFor(x => x.DateFrom)
                              .Must(date =>  new DateValidator.Validate(dateFrom = date).IsValid)
                              .WithMessage("SomeMsg");
            RuleFor(x => x.DateTo)
                             .Must(date =>  new DateValidator().Validate(dateFrom).IsValid &&  
                                            new DateValidator().Validate(dateTo = date).IsValid)
                             .WithMessage("SomeMsg");
        }
    }
```

**Reply by mishael-o** [reply]
Hi @Sharaf-Mansour, thanks for the suggestion, I will check it out.
